packages/nuxt/src/app/composables/fetch.ts

```diff
@@ -47,7 +47,7 @@ export function useFetch<
   arg2?: string
 ) {
   const [opts = {}, autoKey] = typeof arg1 === 'string' ? [{}, arg1] : [arg1, arg2]
-  const _key = opts.key || hash([autoKey, unref(opts.baseURL), typeof request === 'string' ? request : '', unref(opts.params)])
+  const _key = opts.key || hash([autoKey, unref(opts.baseURL), typeof request === 'string' ? request : '', unref(opts.params), unref(opts.query)])
```

Comment:
Alternatively, we could also use || to reduce hash complexity (both shouldn't be in same options!

```suggestion
  const _key = opts.key || hash([autoKey, unref(opts.baseURL), typeof request === 'string' ? request : '', unref(opts.params || opts.query)])
```

Reply:
Sure, do you want me to change it?